src/reader.rs

```diff
@@ -97,6 +120,56 @@ fn recurse_files(path: impl AsRef<Path>) -> std::io::Result<Vec<PathBuf>> {
     Ok(buf)
 }
 
+fn is_ifsysctl(file_name: &str) -> bool {
+    let invalid_suffix = [
+        "~",
+        ".old",
+        ".bak",
+        ".orig",
+        ".scpmbackup",
+        ".rpmnew",
+        ".rpmsave",
+        ".rpmorig",
+    ];
+
+    if !file_name.starts_with("ifsysctl") {
+        return false;
+    }
+
+    for suffix in invalid_suffix {
+        if file_name.ends_with(suffix) {
+            return false;
+        }
+    }
+
+    true
+}
+
+fn warn_on_deprecated_ifsysctl() -> Result<(), anyhow::Error> {
+    let settings = MIGRATION_SETTINGS.get().unwrap();
+
+    let entries = read_dir(&settings.netconfig_base_dir)?;
```

Comment:
Since we already have a `recurse_files` helper function maybe it makes sense to also introduce a helper function like `read_files_in_dir`. But not important :slightly_smiling_face:

Reply:
changed `recurse_files` to `list_files(path, recusive)`